src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs

```diff
@@ -49,10 +49,23 @@ public void CheckIsSharedMefComponent()
             MefTestHelpers.CheckIsSingletonMefComponent<SolutionInfoProvider>();
         }
 
+        [TestMethod]
+        public void Ctor_DoesNotCallServices()
```

Comment:
New test. The old code incorrectly fetched a VS service in the constructor. This test makes sure we don't change the behaviour of the constructor in the future without thinking about the implications.